### ISSUE TITLE
Win32 window shouldn't unconditionally intercept keys

### ISSUE THREAD

**Issue by Gutza** [opencv/opencv#20481]
Currently, Win32 window objects [automatically and unconditionally intercept key codes for `Ctrl+C` and `Ctrl+S`](https://github.com/opencv/opencv/blob/85dde8a800fdb8c4b92bd9088174079cc5a4d416/modules/highgui/src/window_w32.cpp#L2224). That's a nice default, but the developers should be able to inhibit this behavior.

I'm developing a Python mini-app which has to save its own files, and `Ctrl+S` is such a standard key combination for saving data that it makes it really awkward, counter-intuitive, and error-prone to have to implement a different key combination for saving. `Ctrl+C` would also be useful for interrupting execution (if you want to interrupt the process running in the console you press `Ctrl+C`, but if you happen to have the window on top you end up copying to clipboard with the same key combination).

I think this could be implemented relatively easily by allowing for one or two more options for the flags in [`cv::namedWindow()`](https://docs.opencv.org/4.5.2/d7/dfc/group__highgui.html#ga5afdf8410934fd099df85c75b2e0888b), but I'm not a cpp developer, and I don't want to mess things up.

**Comment by alalek** [comment]
HighGUI module is designed for easy debug and demo purposes (probably in some cross platform way).
There is no goal to provide full featured replacement of UI libraries in Computer Vision library.
Consider using other UI libraries to provide GUI of application.

**Comment by Gutza** [comment]
Fair enough, that's a legitimate consideration. However, please keep in mind I'm not asking for some crazy complex infrastructure–I'm just saying that something that's (presumably) relatively easy to implement and doesn't break anything is worth considering. The entire library is so flexible, easy to use, and complete that employing external libraries for minor solutions is probably not worth it for most people.
